### PR TITLE
Pipeline + Preprocessing + UDF UT

### PR DESCRIPTION
**Pull request by oualib**
- Correcting Pipelines
 - Correcting One Hot Encoder
 - Correcting Preprocessing techniques
 - Adding the UT for udf

### CODE REVIEW

**Comment by oualib** [comment]
In the next version (0.4.1), we will support the following new features:
 - to_pickle vdf function
 - udf generator
 - het_arch, het_breuschpagan, het_goldfeldquandt, het_white, kurtosistest, normaltest, skewtest, normaltest functions
  - new geo functions
  - new methods in the regression report: anova/details/metrics
  - new metrics: r2 adjusted Root MSE
  - new function anova_table in metrics
 
In the next-next version (0.4.2), we will support:
 - Pipeline (only UT missing)
 - Polynomial transformation
 - more geospatial functions

**Review by afard — COMMENTED** [review]
The best practice is to have new features and their unit tests in the same PR.
Please add unit tests for Pipeline into this PR.

**Review by afard — APPROVED** [review]

File: verticapy/tests/vDataFrame/test_vDF_utilities.py [hunk 1]
```diff
@@ -308,6 +315,7 @@ def test_vDF_to_pandas(self, titanic_vd):
     def test_vDF_to_pickle(self, titanic_vd):
         result = titanic_vd.select(["age", "survived"])[:20].to_pickle("save.p")
         import pickle
+        pickle.DEFAULT_PROTOCOL = 4
         result_tmp = pickle.load(open("save.p", "rb"))
         result_tmp.set_cursor(titanic_vd._VERTICAPY_VARIABLES_["cursor"])
         result_tmp.shape == (20, 2)
```
[the remark this answers is not in the document]
Comment (afard) [reply verticapy/tests/vDataFrame/test_vDF_utilities.py]: Actually, I added assert here, but then it made the test fail. I leave it to you to figure it out.

**Review by afard — COMMENTED** [review]
Please fix the assert in test_vDF_to_pickle.

**Comment by oualib** [comment]
Done;

**Pull request merged**